src/__tests__/metrics.test.ts

```diff
@@ -8,21 +8,46 @@ const params = {
 }
 
 describe('metrics', () => {
-  test.each([[counter, Counter], [histogram, Histogram], [gauge, Gauge], [summary, Summary]])(
-    '%p passes correct params to %p',
-    (a, b) => {
-      a(params)
-      expect(b).toHaveBeenCalledWith(params)
-    }
-  )
+  test.each([
+    [counter, Counter],
+    [histogram, Histogram],
+    [gauge, Gauge],
+    [summary, Summary]
+  ])('%p passes correct params to %p and prevents register duplicate metrics', (a, b) => {
+    // @ts-expect-error (ts2339)
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+    register.getSingleMetric.mockReturnValue(undefined)
+    const result1 = a(params)
+    expect(b).toHaveBeenCalledWith(params)
+    // eslint-disable-next-line @typescript-eslint/unbound-method
+    expect(register.getSingleMetric).toHaveBeenCalledTimes(1)
+    // eslint-disable-next-line @typescript-eslint/unbound-method
+    expect(register.getSingleMetric).toHaveBeenLastCalledWith(params.name)
 
-  it('collects metrics', () => {
-    collect()
+    // @ts-expect-error (ts2339)
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-call
+    register.getSingleMetric.mockReturnValue(result1)
+    const result2 = a(params)
+    expect(b).toHaveBeenCalledWith(params)
+    // eslint-disable-next-line @typescript-eslint/unbound-method
+    expect(register.getSingleMetric).toHaveBeenCalledTimes(2)
+    // eslint-disable-next-line @typescript-eslint/unbound-method
+    expect(register.getSingleMetric).toHaveBeenLastCalledWith(params.name)
+    expect(result1).toEqual(result2)
+  })
+
+  it('collects metrics', async () => {
+    await collect()
+    // eslint-disable-next-line @typescript-eslint/unbound-method
```

Comment:
why do we need to add these?

Reply:
Lint runs only for files with changes. It was failing for this